### PR TITLE
DOP-3708: scroll heading into view if hash exists in URL

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-3708

### Current Behavior:

[server docs with anchor/hash in URL](https://www.mongodb.com/docs/manual/tutorial/query-documents/#specify-and-conditions)

### Staging Links:

[staged change with anchor/hash in URL](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/docs/seung.park/DOP-3708/tutorial/query-documents/index.html#specify-and-conditions)

### Notes:
- There are considerably two scroll behaviors happening:
- 1 When the browser loads the SSR'ed page and scrolls to the element found (browser native function)
- 2 When the component loads and calls the newly added function

- adding a DO NOT MERGE label as this is a [SPIKE] ticket

### CODE REVIEW

File: src/components/Heading.js [hunk 1]
```diff
@@ -42,6 +44,22 @@ const Heading = ({ sectionDepth, nodeData, page, ...rest }) => {
   const hasSelectors = selectors && Object.keys(selectors).length > 0;
   const shouldShowMobileHeader = !!(isPageTitle && isTabletOrMobile && hasSelectors);
 
+  const { hash } = useLocation();
+  const headingRef = useRef();
+  const bannerOffset = parseInt(
```
Comment (rayangler) [inline src/components/Heading.js:49]: We have a `theme.size.stripUnit` function that might accomplish the same logic here. Not sure if that helps!

**Review by rayangler — COMMENTED** [review]
I think this solution makes sense, given the circumstances. It just seems like a limitation of tabbed content

File: src/components/Heading.js (hunk 1 above)
Comment (seungpark) [reply src/components/Heading.js:49]: realized the banner and navbar is already taken into account in the offsetTop so removed! but good find on the `stripUnit` fn thank you

File: src/components/Heading.js [hunk 2]
```diff
@@ -42,6 +44,22 @@ const Heading = ({ sectionDepth, nodeData, page, ...rest }) => {
...
+    theme.header.navbarScrollOffset.slice(0, theme.header.navbarScrollOffset.indexOf('px'))
+  );
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/components/Heading.js:53]: great call! moved to a custom hook and used it first on permalinks (seems the most necessary place to have it)

File: src/hooks/use-hash-anchor.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+const useHashAnchor = (id, ref) => {
+  const { hash } = useLocation();
+
+  useEffect(() => {
+    const hashId = hash?.slice(1);
+    if (!hash || id !== hashId) {
+      return;
+    }
+    setTimeout(() => {
+      console.log('scrolling');
```
Comment (rayangler) [inline src/hooks/use-hash-anchor.js:13]: We can remove this

**Review by rayangler — APPROVED** [review]
Cool, seems like it works well! Thanks for working on this

**Pull request merged**